include/ccf/service/node_info.h

```diff
@@ -56,6 +56,10 @@ namespace ccf
     /// Public key
     std::optional<crypto::Pem> public_key = std::nullopt;
 
+    /// Free-form user data, can be used to store operator correlation
+    /// IDs/labels for the node for example
+    nlohmann::json node_data = nullptr;
```

Comment:
Yes, that's necessary for ledger compatibility, good catch.

Reply:
That was discussed here: https://github.com/microsoft/CCF/discussions/2916

We said:
> The new field will either have to be defined as i) std::optional or ii) _a non-std::optional type, with a sensible default value_.

In general, we need to make new fields `OPTIONAL` in the serialisation macro (to deserialise old entries where it is not present), but only need `std::optional` where we need a special value to distinguish cases where it is unset. I understand why we need to do that for the fields above (we expect all new nodes to have a `public_key`, and take different execution paths when that is the case).

But for this `node_data` field, we a) already have a sensible distinct default value (JSON `null`), and b) can treat old and new nodes identically. We never need to distinguish "an old node without `node_data` (because it didn't exist)" from "a new node without `node_data` (because it wasn't set)", and in fact have a third case of "an old node _with_ `node_data` (because it was set on a new peer!)".

So I think this is fine, and fits in our previously discussed restrictions.

Reply:
Ok, this is consistent with what we do for members and users too (using json `null` as the default I mean).